src/services/httpClient.ts

```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
+import { API_BASE_URL } from "@/constants/apiConfig";
+import { ERROR_MESSAGES } from "@/constants/errorMessages";
+
+const appRequest = axios.create({
+  baseURL: API_BASE_URL,
+  timeout: 30 * 1000,
+  responseType: "json",
+  headers: {
+    "Content-Type": "application/json"
+  }
+});
+
+// handle response
+appRequest.interceptors.response.use(
+  res => {
```

Comment:
Did you guys have settings similar to .eslintrc.json? When using TypeScript, strict typing and data validation should be enabled. I noticed several places where strict request parameter types and return data types are needed.
```suggestion
  (res: AxiosResponse) => {
```